src/q5/tdg-INMA1691/summary/tdg-INMA1691-summary.tex

```diff
@@ -0,0 +1,2701 @@
+	&\vdotswithin{=}\\
+	M_{k+1} &= I \oplus (M_k \otimes A)\,.
+	\end{align*}
+	On a fini de converger quand $M_k = M_{k+1}$.
+	L'Algorithme de Dijkstra peut être vu
+	comme une manière efficace d'implémenter cette itération.
+\end{myform}
+
+\subsection{Arbres sous-tendants et énumération}
+\begin{mydef}[Arbre]
+	Un \emph{arbre} est un graphe connexe sans cycle.
+	Une \emph{forêt} est un graphe sans cycle.
+\end{mydef}
+\begin{mydef}[Sous-graphe sous-tendant]
+	Un \emph{sous-graphe sous-tendant} d'un graphe $G$
+	est un sous-graphe qui contient tous les sommets de $G$.
+\end{mydef}
+
+\begin{mytheo}
+	\label{theo:spanning_tree}
+	Tout graphe connexe contient un arbre sous-tendant.
+\end{mytheo}
+\begin{proof}
+Soit $G$ un graphe connexe.
+Parmi tous les sous-graphes sous-tendants connexes de $G$,
+on choisit un sous-graphe $G'$,
+minimal pour l'inclusion dans cet ensemble.
+Pour que $G'$ soit un arbre,
+il faut qu'il soit connexe et acyclique.
+\begin{itemize}
+	\item Il est connexe par construction.
+	\item Supposons que $G'$ ait un cycle.
+	Soit $e = uv$ une arête du cycle.
+	Si on enlève $e$,
+	le graphe est toujours connexe.
+	Pour tout parcours $x \ldots uv \ldots y$,
+	on peut remplacer $e$ par le reste du cycle.
+	\contradiction
+\end{itemize}
+\end{proof}
+
+\begin{mylem}[Feuilles d'un arbre]
+	\label{lem:leaves}
+	Tout arbre avec au moins deux sommets
+	a aux moins deux feuilles (n\oe{}uds de degré un)
+	et supprimer une feuille d'un arbre à $n$
+	n\oe{}uds produit un arbre à $n-1$ n\oe{}uds.
+\end{mylem}
+\begin{proof}
+	Dans un graphe acyclique avec au moins deux sommets,
+	les extrémités d'un chemin maximal n'ont pas d'autres voisins
+	que leurs voisins sur le chemin.
+	Ces extrémités sont donc des feuilles.
+
+	Soit $v$ une feuille d'un arbre $G = (V, E, \phi)$.
+	Soit $G' = (V', E', \phi') = G - v$.
+	Comme un sommet de degré un
+	n'appartient à aucun chemin reliant deux autres sommets,
+	pour tout $u,v \in V'$,
+	tout chemin de $u$ à $v$ dans $G$ est aussi dans $G'$.
+	$G'$ est donc connexe.
+
+	Puisque supprimer un sommet ne peut pas créer de cycle,
+	$G'$ est également acyclique.
+\end{proof}
+
+\subsubsection{Caractérisations des arbres}
+\begin{mytheo}[Caractérisations des arbres]
+	Soit $G$ un graphe à $n$ sommets et $m$ arêtes.
+	Alors les conditions suivantes sont équivalentes.
+	\begin{enumerate}[label=(\arabic*)]
+		\item \label{conn_acy} $G$ est connexe et sans cycle.
+		\item \label{acy_mn1} $G$ est sans cycle et $m = n - 1$.
+		\item \label{conn_mn1} $G$ est connexe et $m = n-1$.
+		\item \label{conn_del}$G$ est connexe
+		et supprimer une arête quelconque déconnecte $G$.
+		\item \label{acy_add} $G$ est sans cycle
+		et ajouter une arête quelconque crée un et un seul cycle.
+		\item \label{link} Deux n\oe{}uds de $G$ sont toujours reliés
+		par un seul chemin.
+	\end{enumerate}
+	La dernière condition implique que $G$ est sans boucle
+	(pour deux n\oe{}uds identiques).
+\end{mytheo}
+\begin{proof}
+	\noindent
```

Comment:
On me sort un "underfull \hbox in paragraph" ici ;)

Reply:
C'est le double `\newline`, mais j'en ai besoin sinon la `\fbox` va se coller contre le "Démonstration"...